### ISSUE TITLE
Coinbase output script in pool conf not as easy as using address directly

### ISSUE THREAD

**Issue by rrybarczyk** [stratum-mining/stratum#1199]
# Background
The `roles::pool` config toml accepts coinbase outputs by accepting a `output_script_type` and ` output_script_value`. This `output_script_value` is the pool's compressed public key.

# Problem
After speaking with @russeree, he mentioned it would be easier to support a coinbase address directly here. That on first look

# Solution
Add support for a coinbase address in the pool config and logic.

**Comment by rrybarczyk** [comment]
Do we want to continue to support the `output_script_value`? I would say yes for now, and add the coinbase address as an additional alternative option. In the future, if we see that `output_script_value` is not used, then could consider deprecating.

**Comment by GitGab19** [comment]
I agree it's a bit tricky to use what we have now 👍  It caused confusion also for some people who tried the `benchmarking-tool`, so we should work on it.
In the meantime, I would say that we have [this crate](https://github.com/stratum-mining/stratum/tree/main/utils/bip32-key-derivation) to "easily" get the pubkey from a BIP32 xpub if it can help.

**Comment by russeree** [comment]
I believe pubkey to address encoding is done here

https://github.com/stratum-mining/stratum/blob/main/protocols/v2/roles-logic-sv2/src/utils.rs#L191

Could we just use the rust bitcoin lib to fetch the address type? The use the helper functions to return the output script? This should let the user just place in an address and the process of getting the script from that would be handled. 


Helpers that could be useful

- https://docs.rs/bitcoin/latest/bitcoin/address/struct.Address.html#method.address_type
- https://docs.rs/bitcoin/latest/bitcoin/address/enum.AddressData.html
- https://docs.rs/bitcoin/latest/bitcoin/address/struct.Address.html#method.pubkey_hash
- https://docs.rs/bitcoin/latest/bitcoin/address/struct.Address.html#method.p2tr

If this is a reasonable path, I can pull this.

**Comment by GitGab19** [comment]
Yes now the encoding is done there in the `try_from`. 
Your approach looks good to me 👍
